napari/_qt/_qapp_model/qactions/_plugins.py

```diff
@@ -0,0 +1,73 @@
+"""Defines plugins menu q actions."""
+
+from importlib.util import find_spec
+from logging import getLogger
+from typing import List
+
+from app_model.types import Action
+
+from napari._app_model.constants import CommandId, MenuGroup, MenuId
+from napari._qt.dialogs.qt_plugin_report import QtPluginErrReporter
+from napari._qt.qt_main_window import Window
+from napari.utils.translations import trans
+
+logger = getLogger(__name__)
+
+
+def _plugin_manager_dialog_avail() -> bool:
+    """Returns whether the plugin manager class is available."""
+
+    plugin_dlg = find_spec('napari_plugin_manager')
+    if plugin_dlg:
+        return True
+    # not available
+    logger.debug("QtPluginDialog not available")
+    return False
+
+
+def _show_plugin_install_dialog(window: Window) -> None:
+    """Show dialog that allows users to install and enable/disable plugins."""
+
+    # TODO: Once menu contributions supported, `napari_plugin_manager` should be
+    # amended to be a napari plugin and simply add this menu item itself.
+    # This callback is only used when this package is available, thus we do not check
+    from napari_plugin_manager.qt_plugin_dialog import QtPluginDialog
+
+    QtPluginDialog(window._qt_window).exec_()
+
+
+def _show_plugin_err_reporter(window: Window) -> None:
+    """Show dialog that allows users to review and report plugin errors."""
+    QtPluginErrReporter(parent=window._qt_window).exec_()  # type: ignore [attr-defined]
```

Comment:
Just wanted to highlight that mypy did not recognize `exec_` as a attribute, thus the ignore comment. Hope this is fine.

Reply:
Yeah I don't know but double checked and it's definitely arises:

https://github.com/napari/napari/actions/runs/6952876782/job/18917143401?pr=4991

```
 napari/_qt/_qapp_model/qactions/_plugins.py:41: error: "QtPluginErrReporter" has no attribute "exec_"  [attr-defined]
```

Reply:
Okay I could not leave this alone. This was fine before you moved it because it is ignored in `pyproject.toml`:
https://github.com/napari/napari/blob/85c58b5990dfbbc41468ae16c25216e8a00f11a3/pyproject.toml#L496-L519

So, I think it's probably fine to leave the ignore comment there. However I also found that the [Qt docs for `QDialog.exec()`](https://doc.qt.io/qt-6/qdialog.html#exec) suggest using `open()` instead:
> Note: Avoid using this function; instead, use open(). Unlike exec(), [open](https://doc.qt.io/qt-6/qdialog.html#open)() is asynchronous, and does not spin an additional event loop. This prevents a series of dangerous bugs from happening (e.g. deleting the dialog's parent while the dialog is open via exec()). When using [open](https://doc.qt.io/qt-6/qdialog.html#open)() you can connect to the [finished](https://doc.qt.io/qt-6/qdialog.html#finished)() signal of [QDialog](https://doc.qt.io/qt-6/qdialog.html) to be notified when the dialog is closed.

Changing this code to use `open()` seems to work for me (functionality remains and mypy stops complaining), but I am not familiar enough to know if there is a reason we are using `exec` instead.

Reply:
That doc is pretty explicit lol. @lucyleeow maybe let's leave it like this for now in the interest of minimal change, but follow up with an issue about using `open()` instead? I dunno, whatever you think is best.

Reply:
Yes I'll open a new issue for it, I guess it's better for git blame purposes. This is already a pretty big PR. I'll also have a quick dig to see the history of our `QtPluginErrReporter.exec_` use.
Thanks for the detective work @aganders3 :detective: !!!

Reply:
#6502 done